File: api/src/modules/shop/checker/checks/worker.ts
import type { Checker, CheckerInput, CheckerOutput } from '../registery.ts';

export default class implements Checker {
    async check(input: CheckerInput, result: CheckerOutput): Promise<void> {
        if (input.config.adminWorker.enableAdminWorker) {
            result.warning(
                'admin.worker',
                'Admin-Worker should be disabled',
                'Shopmon',
                'https://developer.shopware.com/docs/guides/plugins/plugins/framework/message-queue/add-message-handler#the-admin-worker',
            );
            return;
        }

        result.success('admin.worker', 'Admin-Worker is disabled');
    }
}

File: api/internal/shopware/checker/worker.go
package checker

import "context"

func checkWorker(_ context.Context, input Input, output *Output) {
	if input.Config.AdminWorker.EnableAdminWorker {
		output.Warning("admin.worker", "check.worker.enabled", nil, "Shopware",
			"https://developer.shopware.com/docs/guides/hosting/infrastructure/message-queue.html")
	} else {
		output.Success("admin.worker", "check.worker.disabled", nil, "Shopware", "")
	}
}
